### PR TITLE
[Doc Improvement][Add new status codes for bot conversational APIs][2678372]

### PR DESCRIPTION
**Pull request by v-npaladugu**

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/9a5eb5e0d6f891c1e70fa268f8806c86fa98ddd5/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 89 | 25 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/b3812329-1dc9-4a24-809c-5be7ff4fa3fc) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/544894ed073f21b60560fbac6fd68c7db8a98a88/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 90 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3aab08ae-6dd8-497a-8db1-86280c09c195) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/eca9b76afbc0b7d0dd9bd09abd3d03da6b2a0789/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 90 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/41bbadcd-8ea5-4c5f-858a-b7ad4f9e2102) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by v-habibaumme — APPROVED** [review]
Approved

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 1]
```diff
@@ -467,6 +468,7 @@ The general retry guidance for each status code is listed in the following table
 
 |Status code | Retry strategy |
 |----------------|-----------------|
+| 403 |Retry by calling the GCC API `https://smba.infra.gcc.teams.microsoft.com`.|
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:471]: We should clarify that this applies only if the code is InvalidBotApiHost.

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 2]
```diff
@@ -448,8 +448,9 @@ Ensure to handle these errors appropriately in your Teams app. The following tab
...
+| 403 | **Code**: `BotNotInConversationRoster` <br/> **Message**: The bot isn't part of the conversation roster. | The bot isn't part of the conversation. App needs to be reinstalled in conversation. | No | Before attempting to send another conversation requests, wait for an [`installationUpdate`](~/bots/how-to/conversations/subscribe-to-conversation-events.md#install-update-event) event, which indicates that the bot has been re-added.|
 | 403 | **Code**: `ConversationBlockedByUser` <br/> **Message**: User blocked the conversation with the bot. | User has blocked the bot in personal chat or a channel through moderation settings. | No | Delete the conversation from cache. Stop attempting to post to conversations until interaction with bot is explicitly initiated by a user in the conversation, indicating that the bot is no longer blocked. |
+| 403 |**Code**: `InvalidBotApiHost` <br/> **Message**: Invalid bot api host. For GCC tenants please call `https://smba.infra.gcc.teams.microsoft.com`.|Accessed public API from GCC tenant.|Yes|Retry by calling the GCC API `https://smba.infra.gcc.teams.microsoft.com`|
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:453]: > Accessed public API from GCC tenant.

Perhaps "The bot called the public API endpoint for a conversation that belongs to a GCC tenant." to clarify exactly what is wrong

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 3]
```diff
@@ -448,8 +448,9 @@ Ensure to handle these errors appropriately in your Teams app. The following tab
...
-| 403 | **Code**: `BotNotInConversationRoster` <br/> **Message**: The bot isn't part of the conversation roster. | The bot isn't part of the conversation. App needs to be reinstalled in conversation. | No | Before attempting to send additional conversation requests, wait for an [`installationUpdate`](~/bots/how-to/conversations/subscribe-to-conversation-events.md#install-update-event) event, which indicates that the bot has been re-added.|
+| 403 | **Code**: `BotNotInConversationRoster` <br/> **Message**: The bot isn't part of the conversation roster. | The bot isn't part of the conversation. App needs to be reinstalled in conversation. | No | Before attempting to send another conversation requests, wait for an [`installationUpdate`](~/bots/how-to/conversations/subscribe-to-conversation-events.md#install-update-event) event, which indicates that the bot has been re-added.|
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:451]: > another conversation requests

Nit: "another conversation request" or "another request for this conversation"
(change from additional->another requires the noun to change from plural->singular)

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ Each message in a conversation is an `Activity` object of type `messageType: mes
 Basic conversations are handled through the Bot Framework connector, a single REST API. This API enables your bot to communicate with Teams and other channels. The Bot Builder SDK provides the following features:
 
 * Easy access to the Bot Framework connector.
-* Additional functionality to manage conversation flow and state.
+* Another functionality to manage conversation flow and state.
```
Comment (aosolis) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:16]: Why did we make this change? The new phrasing sounds awkward, maybe omit "Another" entirely and just say "Functionality to manage conversation flow and state"?

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/ff4a6a045bb07b66ae4364094bd6c4e26e114595/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 90 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/166a1596-1d14-4483-bd8c-c539ba8e701e) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/bots/how-to/conversations/conversation-messages.md (hunk 4 above)
Comment (surbhigupta12) [reply msteams-platform/bots/how-to/conversations/conversation-messages.md:16]: @v-npaladugu,

Additional is the right word. Please revert. Kindly, gte the PRs reviewed by me before we reach out to SMEs.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/1775f0a3d8856067478e775d7975359e3dd86688/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 90 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/054cd042-0f78-447d-9336-30c3b3f11cb3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/d5c47baa8637b8a389b8672645b7944533b15d74/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 90 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/17611365-89ef-46ef-9177-e74fe3409469) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 5]
```diff
@@ -448,8 +448,9 @@ Ensure to handle these errors appropriately in your Teams app. The following tab
...
+| 403 | **Code**: `BotNotInConversationRoster` <br/> **Message**: The bot isn't part of the conversation roster. | The bot isn't part of the conversation. App needs to be reinstalled in conversation. | No | Before attempting to send another conversation request, wait for an [`installationUpdate`](~/bots/how-to/conversations/subscribe-to-conversation-events.md#install-update-event) event, which indicates that the bot has been re-added.|
 | 403 | **Code**: `ConversationBlockedByUser` <br/> **Message**: User blocked the conversation with the bot. | User has blocked the bot in personal chat or a channel through moderation settings. | No | Delete the conversation from cache. Stop attempting to post to conversations until interaction with bot is explicitly initiated by a user in the conversation, indicating that the bot is no longer blocked. |
+| 403 |**Code**: `InvalidBotApiHost` <br/> **Message**: Invalid bot api host. For GCC tenants please call `https://smba.infra.gcc.teams.microsoft.com`.|The bot called the public API endpoint for a conversation that belongs to a GCC tenant.|Yes|Retry by calling the GCC API `https://smba.infra.gcc.teams.microsoft.com`.|
```
Comment (tacri) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:453]: I think retry should be "No" - I can see someone just going ahead and coding the retries from here without reading the entire line and if they retry with no change the request will fail. In the right column I would say "Update the service url for this conversation to https://smba.infra.gcc.teams.microsoft.com and retry the request.

**Review by leonyang12 — APPROVED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/bots/how-to/conversations/conversation-messages.md](https://github.com/MicrosoftDocs/msteams-docs/blob/d82631b6982917c9869b8d721b3f1a6e375ea3a3/msteams-platform%2Fbots%2Fhow-to%2Fconversations%2Fconversation-messages.md) | 92 | 21 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/22323de2-aa7e-4ed3-a1a6-a9663f6ccd92) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by surbhigupta12 — APPROVED** [review]

**Pull request merged**